pelican/action.yml

```diff
@@ -51,34 +51,55 @@ runs:
     # If the site uses Github Flavored Markdown, use this build branch
     - name: fetch and build libcmark-gfm.so
       if: ${{ inputs.gfm == 'true' }}
+      id: build_gfm
       shell: bash
       env:
-        WORKDIR: /opt/pelican-asf # where to build GFM
         GFM_VERSION: '0.28.3.gfm.12' # ensure we agree with build-cmark.sh script
+        # action_repository only works in the env context; empty for local action call
+        # it is always empty for local invocation, in which case use the current repo
+        GITHUB_ACTION_REPO: ${{ github.action_repository || github.repository }}
+        GH_TOKEN: ${{ github.token }} # needed by gh
       run: |
+        # The key needs to include the GFM version, but is otherwise arbitrary.
+        # It must agree with the definition in build-actions.yml
+        export GFM_ARTIFACT_KEY=gfm-lib-${GFM_VERSION}
+
+        if [[ -z $LIBCMARKDIR ]] # define LIBCMARKDIR if it is not already
+        then
+          # set up the GFM environment
+          export LIBCMARKDIR=/opt/pelican-asf/gfm-${GFM_VERSION} # arbitrary, but should contain version
+          mkdir -p $LIBCMARKDIR
+          echo  "LIBCMARKDIR=${LIBCMARKDIR}" >>$GITHUB_ENV # needed for the build step
+        fi
+
         # Does the GFM build already exist?
-        if [[ -n $LIBCMARKDIR && -d $LIBCMARKDIR ]]
+        if [[ -f $LIBCMARKDIR/libcmark-gfm.so ]]
         then
           echo "Already have GFM binary at $LIBCMARKDIR, skipping build"
           exit 0 # nothing more to do in this step
         fi
+
+        # Is there a saved artifact for the GFM build?
+        echo "Check for GFM build artifact in action repo: $GITHUB_ACTION_REPO"
+        gh run download --dir ${LIBCMARKDIR} --name ${GFM_ARTIFACT_KEY} --repo $GITHUB_ACTION_REPO || true
+        if [[ -f $LIBCMARKDIR/libcmark-gfm.so ]]
+        then
+          echo "Downloaded to ${LIBCMARKDIR} from $GITHUB_ACTION_REPO, nothing more to do!"
+          exit 0 # nothing more to do in this step
+        fi
+
+        # GFM binary not found, need to build it
         {
+          echo "Creating GFM binary in ${LIBCMARKDIR}"
           # disable stdout unless debug is on
           if [ "${{ inputs.debug }}" == 'true' ]
```

Comment:
it's nice if you automatically debug on rerun...

```
|| [ "$GITHUB_RUN_ATTEMPT" != 1 ]
```

Reply:
Note that you should also move this to an `env:`, it isn't just `${{ matrix`, it's `${{` _anything_

Reply:
for most of those, github has a $GITHUB_..., so you really should never do it :)... it's a safe enough rule.